docs/adr/0003-deletion-strategy.md

```diff
@@ -0,0 +1,28 @@
+# 3. Deletion strategy
+
+* Status: proposed
+* Date: 2023-08-22
+* Authors: @furkatgofurov7
+* Deciders: @richardcase @alexander-demicev @salasberryfin @Danil-Grigorev @mjura
+
+## Context
+
+As operator, it is crucial to establish clear guidelines for how Rancher turtles handle cluster deletion operations. The objective is to facilitate the removal of a CAPI-imported cluster from Rancher, ensuring that it no longer appears in the Rancher UI, all while keeping the underlying CAPI cluster intact.
+
+## Decision
+
+The deletion workflow is structured as follows:
+
+**Ownership Chain:** Rancher cluster is associated with CAPI cluster through the owner references chain during their creation process.
+
+**Cluster Annotation:** When deleting a Rancher cluster, the operator will annotate the corresponding CAPI cluster with the `ClusterImportedAnnotation` annotation. This annotation will prevent automatic re-import of the CAPI cluster, only if the Rancher cluster itself contains the same annotation.
+
+From an end-user perspective:
+
+* If user manually deletes CAPI cluster. Rancher turtles uses [Kubernetes owner references](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) to track relationships between objects. These references are used for Kubernetes garbage collection, which is the basis of Rancher cluster deletion in Rancher turtles.
+* If user manually removes the Rancher cluster from UI. Depending on annotation presence in CAPI cluster, operator decides to allow or block importing. 
+
+## Consequences
+
+- The operator will leverage Kubernetes' native garbage collection mechanism, by using the owner reference chain, to ensure a streamlined cluster deletion process.
```

Comment:
Not sure if it is will be possible when running rancher turtles outside of rancher manager cluster, we will have to find a suitable alternative later.

Reply:
This ADR is based on the current state of the operator, we would need to update it probably accordingly later when we can run rancher turtles outside of rancher manager cluster